lms/djangoapps/api_manager/users/views.py

```diff
@@ -916,6 +916,35 @@ def delete(self, request, user_id, course_id):
         return Response({}, status=status.HTTP_204_NO_CONTENT)
 
 
+class UsersCoursesGradesList(SecureAPIView):
+    """
+    ### The UsersCoursesGradesList view allows clients to interact with the User's gradebook across courses
+    - URI: ```/api/users/{user_id}/courses/grades?courses={course_id1,course_id2}```
+    - GET: Returns a JSON representation of the matched set from the gradebook
+        * courses (optional): Set filtering parameter consisting of comma-separated course identifiers
+
+    ### Use Cases/Notes:
+    * Use the UsersCoursesGradesList view to interact with the User's gradebook across multiple Course enrollments
+    * Use GET to retrieve all of the Course gradebook entries for the specified User
+    """
+    def get(self, request, user_id):  # pylint: disable=unused-argument
+        """
+        GET /api/users/{user_id}/courses/grades
+        """
+        queryset = StudentGradebook.objects.filter(user=user_id)
+
+        response_data = []
+        for record in queryset:
+            response_data.append(
+                {
+                    'course_id': unicode(record.course_id),
+                    'current_grade': record.grade,
+                    'proforma_grade': record.proforma_grade,
+                }
```

Comment:
We would also need a boolean value that indicates whether a user completed a course of not. I left you a comment in the line where that implementation was added. You might want to refactor that into the Gradebook so that `complete_status` value is assigned from within the Gradebook and retrieved in user, course or ogranization APIs for further use, instead of being calculated in all of those places. Just a suggestion, you might have a better, more elegant solution for that.